beater/handlers.go

```diff
@@ -133,6 +134,18 @@ func isAuthorized(req *http.Request, secretToken string) bool {
 	return subtle.ConstantTimeCompare([]byte(parts[1]), []byte(secretToken)) == 1
 }
 
+func optionsHandler(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "OPTIONS" {
+			w.Header().Set("Access-Control-Allow-Origin", "*")
+			w.Header().Set("Access-Control-Allow-Methods", "POST")
```

Comment:
@jahtalab Do we need the `Access-Control-Allow-Headers` as well? What should it include?

Reply:
@jahtalab Did you investigate it we can get rid of the preflight OPTIONS all together?

Reply:
hmm i probably need to allow header `Content-Type`...

Reply:
@roncohen ,
We can't get rid of preflight request, since `Content-Type:application/json` is not considered [cors safelisted request header](https://fetch.spec.whatwg.org/#cors-safelisted-request-header). We'd have to switch to one of `application/x-www-form-urlencoded`, `multipart/form-data`, or `text/plain` to get rid of OPTIONS request.

To recap:
Response to OPTIONS should include these 
`Access-Control-Allow-Origin "*";`
`Access-Control-Allow-Headers "Content-Type";`
`Access-Control-Allow-Methods "POST";`

Note: We have to update `Access-Control-Allow-Headers` if we choose to use custom headers on the agent side but for now the above should be enough.

And the response to the original POST request should include:
`Access-Control-Allow-Origin "*";`

In production mode `Access-Control-Allow-Origin` should be configured by the user to something other than "*"

Docs:
https://fetch.spec.whatwg.org/
https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS

Reply:
@jahtalab is it worth switching to `multipart/form-data` or `application/x-www-form-urlencoded` to get rid of the OPTIONS request? It could be a lot of requests we'd end up saving

Reply:
OPTIONS can be cached by the browser if i send the max age header, if that is a problem.
not sure what is reasonable... 1 day?

Reply:
Maybe there are special cases when browsers will cache the OPTIONS request, but I never managed to get them to do it. When i tried it out they seem to ignore the max-age header.